### PR TITLE
Add dialogue client generation

### PR DESCRIPTION
**Pull request by ferozco**
## Before this PR

All dialogue client generation lived in an internal repo.

## After this PR
==COMMIT_MSG==
Add dialogue client generation using `--dialogue` flag
==COMMIT_MSG==

Pending to add support for binary params and cookie auth so that our ETEs just work

### CODE REVIEW

**Comment by iamdanfox** [comment]
This is all a copy and paste from the internal repo right? 

(Aside: seems like my hope of being able to have independently released java generators was unrealistic :/)

File: conjure-java-core/src/test/java/com/palantir/conjure/java/DialogueServiceGeneratorTests.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+    @TempDir
+    public File folder;
+
+    private static String compiledFileContent(File srcDir, String clazz) throws IOException {
+        return new String(Files.readAllBytes(Paths.get(srcDir.getPath(), clazz)), StandardCharsets.UTF_8);
+    }
+
+    @Test
+    @Disabled("Pending support for all valid conjure definitions")
```
Comment (iamdanfox) [inline conjure-java-core/src/test/java/com/palantir/conjure/java/DialogueServiceGeneratorTests.java]: :+1: I appreciate not making this a bazillion line PR

**Review by iamdanfox — APPROVED** [review]

**Comment by iamdanfox** [comment]
I've put `no changelog` on this because I don't think we wanna advertise it _just yet_.

**Comment by ferozco** [comment]
👍

**Pull request merged**